lib/bolt/executor.rb

```diff
@@ -26,39 +34,22 @@ def initialize(config = Bolt::Config.new, noop = nil, plan_logging = false)
       @notifier = Bolt::Notifier.new
     end
 
-    def from_targets(targets)
-      targets.map do |target|
-        Bolt::Node.from_target(target)
-      end
-    end
-    private :from_targets
-
-    def on(nodes, callback = nil)
+    def on(targets, callback = nil)
       results = Concurrent::Array.new
 
-      poolsize = [nodes.length, @config[:concurrency]].min
+      poolsize = [targets.length, @config[:concurrency]].min
       pool = Concurrent::FixedThreadPool.new(poolsize)
       @logger.debug { "Started with #{poolsize} thread(s)" }
 
-      nodes.map(&:class).uniq.each do |klass|
-        klass.initialize_transport(@logger)
-      end
-
-      nodes.each { |node|
+      targets.each { |target|
         pool.post do
           result =
             begin
-              @notifier.notify(callback, type: :node_start, target: node.target) if callback
-              node.connect
-              yield node
+              transport = @transports[target.protocol || 'ssh']
+              @notifier.notify(callback, type: :node_start, target: target) if callback
+              yield transport, target
             rescue StandardError => ex
-              Bolt::Result.from_exception(node.target, ex)
-            ensure
-              begin
-                node.disconnect
-              rescue StandardError => ex
-                @logger.info("Failed to close connection to #{node.uri} : #{ex.message}")
-              end
+              Bolt::Result.from_exception(target, ex)
```

Comment:
Now that the diconnect isn't here this will only catch logging/notifying issues right?

Reply:
Yeah, because we run everything wrapped in `with_exception_handling` which does exactly the same thing. So I think this isn't necessary anymore.